reviews/models/mixins.py

```diff
@@ -170,10 +174,28 @@ def update_last_transitioned(self, ev):
     def save_changes(self, ev):
         now = self.action.date_created if self.action is not None else timezone.now()
         should_publish = self.reviewable.in_public_reviews_state
+        user = ev.kwargs.get('user')
+        auth = Auth(user)
         if should_publish and not self.reviewable.is_published:
-            self.reviewable.is_published = True
+            if not (self.reviewable.node.preprint_file and self.reviewable.node.preprint_file.node == self.reviewable.node):
+                raise ValueError('Preprint node is not a valid preprint; cannot publish.')
+            if not self.reviewable.provider:
+                raise ValueError('Preprint provider not specified; cannot publish.')
+            if not self.reviewable.subjects.exists():
+                raise ValueError('Preprint must have at least one subject to be published.')
             self.reviewable.date_published = now
-            # TODO EZID (MOD-70)
+            self.reviewable.node._has_abandoned_preprint = False
+            self.reviewable.is_published = True
+
+            self.reviewable.node.add_log(
+                action=NodeLog.PREPRINT_INITIATED,
```

Comment:
I think it makes more sense to add this log when the preprint is submitted, not when it's published. It might make sense in `notify_submit`?